t/listeners.t

```diff
@@ -0,0 +1,78 @@
+#! perl
+
+use strict;
+use warnings;
+use Test::More;
+use Test::Exception;
+
+use Scalar::Util qw[ refaddr ];
+
+{
+    package Foo;
+    use Moo;
+    with 'Beam::Emitter';
+}
+
+{
+    package Goo;
+    use Moo;
+    extends 'Beam::Listener';
+
+    has attr => ( is => 'ro', required => 1 );
+
+}
+
+sub byref { refaddr $a <= refaddr $b }
+
+my $foo = Foo->new;
+
+my $s11 = sub { 'evt11' };
+my $s12 = sub { 'evt12' };
+my $s21 = sub { 'evt21' };
+my $s22 = sub { 'evt22' };
+
+my $us11 = $foo->subscribe( evt1 => $s11 );
+my $us12 = $foo->subscribe( evt1 => $s12 );
+my $us21 = $foo->subscribe( evt2 => $s21 );
+
+# try an alternate listener class.
+
+# test constructor is being called with args
+throws_ok(
+    sub { $foo->subscribe( evt2 => $s22, class => 'Goo' ) },
+    qr/missing required arguments/i,
+    "missing attr for custom listener class"
+);
+
+my $us22 = $foo->subscribe( evt2 => $s22, class => 'Goo', attr => 's22' );
+
+{
```

Comment:
For each of these naked blocks in the test, could you modify them slightly to use `subtest` and provide a short description of what the test is intended to demonstrate? Something like: `subtest "Ensure the correct listener object is removed from the list" => sub { ... };`. All the Beam stuff relies on a version of Test::More that has `subtest`.